plutus-core/plutus-core/test/Spec.hs

```diff
@@ -191,7 +191,7 @@ propParser = property $ do
                 (\p -> fmap (TextualProgram . void) (parseProg p))
     where
         parseProg :: T.Text -> Either ParserErrorBundle (Program TyName Name DefaultUni DefaultFun SourcePos)
-        parseProg p = runQuoteT $ parseProgram p
+        parseProg = runQuoteT . parseProgram "test"
```

Comment:
We don't actually have a file name at this point, so I just got it to do what it was doing before.

Reply:
My thought was that `parseProgram` is for when we don't use the file name. Otherwise, you should use `parse`. Perhaps we can rename the old `parseProgram` to `parseProgramWithoutName`?

Reply:
Yes, maybe that's a good idea.  Once I started doing this it turned out that `parseProgram` was used in more places than I'd expected, so maybe exporting both a generic one and one that takes a name too is the right way to go.  I'll look at that tomorrow.

Reply:
or just pipe it up further? it's easy enough for the caller to choose a name

Reply:
there are cases that the name just isn't used/needed. For cases where the name is useful, one should use `parse`.

Reply:
@thealmarty
> Perhaps we can rename the old parseProgram to parseProgramWithoutName?
I tried that but it was kind of annoying because we still have `parseTerm` and `parseType` which do similar things and it was inconsistent with them.  I've put `parseProgaram` back to what it was before and changed the name of  the method in `Excutable` to `parseNamedProgram` instead and got it to use `pare program`.  That's the only place where we use the name of the input, so that seems reasonably sensible.